src/views/Lottery/components/WinningNumbers.tsx

```diff
@@ -1,20 +1,50 @@
-import React from 'react'
-import { Flex } from '@pancakeswap/uikit'
+import React, { useEffect, useState } from 'react'
+import { Flex, FlexProps } from '@pancakeswap/uikit'
+import { random } from 'lodash'
+import uniqueId from 'lodash/uniqueId'
 import { parseRetreivedNumber } from '../helpers'
 import { BallWithNumber } from '../svgs'
+import { BallColor } from '../svgs/Balls'
 
-const WinningNumbers: React.FC<{ number: string }> = ({ number }) => {
+interface WinningNumbersProps extends FlexProps {
+  number: string
+  size?: string
+  fontSize?: string
+  rotateText?: boolean
+}
+
+const WinningNumbers: React.FC<WinningNumbersProps> = ({
+  number,
+  size = '32px',
+  fontSize = '16px',
+  rotateText,
+  ...containerProps
+}) => {
+  const [rotationValues, setRotationValues] = useState([])
   const reversedNumber = parseRetreivedNumber(number)
   const numAsArray = reversedNumber.split('')
+  const colors: BallColor[] = ['pink', 'lilac', 'teal', 'aqua', 'green', 'yellow']
+
+  useEffect(() => {
+    if (rotateText && numAsArray && rotationValues.length === 0) {
+      setRotationValues(numAsArray.map(() => random(-30, 30)))
+    }
+  }, [rotateText, numAsArray, rotationValues])
```

Comment:
Changes to allow an optional random rotation to be applied to the balls.

Also dynamically render the balls with a map instead of explicitly declaring them